cpp/merkletree/sparse_merkle_tree.cc

```diff
@@ -126,10 +127,11 @@ string SparseMerkleTree::CalculateSubtreeHash(size_t depth, IndexType index) {
         if (!it->second.hash_.empty()) {
           return it->second.hash_;
         }
-        IndexType left_child_index(index << 1);
+        IndexType left_child_index(index);
+        IndexType right_child_index(index);
+        SetPathBit(&right_child_index, depth + 1, 1);
```

Comment:
To make the prefix keys match the path, rather than be a section of it that's shifted some amount; trying to make it all a bit easier to understand.